cf-net/cf-net.c

```diff
@@ -0,0 +1,766 @@
+    if (conn == NULL)
+    {
+        return 1;
+    }
+    if (print == true)
+    {
+        printf("Connected & authenticated successfully to '%s'\n", server);
+    }
+    DisconnectServer(conn);
+    return 0;
+}
+
+static int CFNetConnect(CFNetOptions *opts, const char* hostname, char **args)
+{
+    if (hostname != NULL)
+    {
+        CFNetConnectSingle(hostname, true); // FIXME
+        return 0;
+    }
+    // TODO
+    if (args == NULL)
+    {
+        return 0;
+    }
+    if (opts->used_default)
+    {
+        return 0;
+    }
+    for (int i = 0; args[i] != NULL; ++i)
+    {
+        CFNetConnectSingle(args[i], true);
+    }
+    return 0;
+}
+
+static void CFNetStatPrint(const char *file, int st_mode, const char *server)
+{
+    printf("%s:", server);
+    if (S_ISDIR(st_mode))
+    {
+        printf("'%s' is a directory\n", file);
+    }
+    else if (S_ISREG(st_mode))
+    {
+        printf("'%s' is a regular file\n", file);
+    }
+    else if (S_ISSOCK(st_mode))
+    {
+        printf("'%s' is a socket\n", file);
+    }
+    else if (S_ISCHR(st_mode))
+    {
+        printf("'%s' is a character device file\n", file);
+    }
+    else if (S_ISBLK(st_mode))
+    {
+        printf("'%s' is a block device file\n", file);
+    }
+    else if (S_ISFIFO(st_mode))
+    {
+        printf("'%s' is a named pipe (FIFO)\n", file);
+    }
+    else if (S_ISLNK(st_mode))
+    {
+        printf("'%s' is a symbolic link\n", file);
+    }
+    else
+    {
+        printf("'%s' has an unrecognized st_mode\n", file);
+    }
+}
+
+static int CFNetStat(CFNetOptions *opts, const char* hostname, char **args)
+{
+    assert(opts);
+    char *file = args[0];
+    AgentConnection *conn = CFNetOpenConnection(hostname);// FIXME
+    if (conn == NULL)
+    {
+        return 1;
+    }
+    bool encrypt = true;
+    struct stat sb;
+    int r = cf_remote_stat(conn, encrypt, file, &sb, "file");
+    if (r!= 0)
+    {
+        printf("Could not stat: '%s'\n", file);
+    }
+    else
+    {
+        Log(LOG_LEVEL_INFO, "Detailed stat output:\n"
+                            "mode  = %jo, \tsize = %jd,\n"
+                            "uid   = %ju, \tgid = %ju,\n"
+                            "atime = %jd, \tmtime = %jd\n",
+            (uintmax_t) sb.st_mode,  (intmax_t)  sb.st_size,
+            (uintmax_t) sb.st_uid,   (uintmax_t) sb.st_gid,
+            (intmax_t)  sb.st_atime, (intmax_t)  sb.st_mtime);
+        CFNetStatPrint(file, sb.st_mode, hostname);
+    }
+
+    DisconnectServer(conn);
+    return 0;
+}
+
+static int invalid_command(const char *cmd)
+{
+    Log(LOG_LEVEL_ERR, "Invalid, see: 'cf-net help %s' for more info\n", cmd);
+    return -1;
+}
+
+
+static int CFNetGet(CFNetOptions *opts, const char* hostname, char **args)
+{
+    assert(opts);
+    assert(hostname);
+    assert(args);
+    AgentConnection *conn = CFNetOpenConnection(hostname);
+    if (conn == NULL)
+    {
+        return 1;
```

Comment:
If you're following POSIX style return type, using `int`, I would consider returning -1 here instead of 1. Although personally I would opt for `bool` return type in most cases. No strong preference, but just pointing it out for consistency's sake; I'll leave the decision to you.

There are many other instances of `return 1` as well.